lib/Alchemy/Phrasea/Core/Event/Subscriber/CollectionSubscriber.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of Phraseanet
+ *
+ * (c) 2005-2014 Alchemy
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Alchemy\Phrasea\Core\Event\Subscriber;
+
+use Alchemy\Phrasea\Core\Event\CollectionEvent\ChangeNameEvent;
+use Alchemy\Phrasea\Core\Event\CollectionEvent\CollectionIndexEvent;
+use Alchemy\Phrasea\Core\Event\DataboxEvent\IndexEvent;
+use Alchemy\Phrasea\Core\PhraseaEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class CollectionSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            PhraseaEvents::COLLECTION_CHANGE_NAME => 'onChangeName',
+        ];
+    }
+
+    public function onChangeName(ChangeNameEvent $event)
+    {
+        $collection = $event->getCollection();
+
+        $dispatcher = $event->getDispatcher();
+        $dispatcher->dispatch(PhraseaEvents::INDEX_COLLECTION, new CollectionIndexEvent($collection));
```

Comment:
I don't like much this part, events should not be used like command IMO. An event is about something that occurred in the past.
The code executing the index process is the best place to know on which events it have to react. Proxying `onChangeName` to `indexCollection` doesn't add value. In fact, we are losing the information of what happened.
This is the indexer which should eventually dispatch an `indexCollection` event when it is doing that. Else why not just have a method call?